src/librustc_resolve/macros.rs

```diff
@@ -782,7 +784,16 @@ impl<'a> Resolver<'a> {
             return Ok(previous_result);
         }
 
-        if record_used { Err(Determinacy::Determined) } else { Err(Determinacy::Undetermined) }
+        let determinacy = Determinacy::determined(force);
+        if determinacy == Determinacy::Determined && is_attr {
+            // For attributes interpret determinate "no solution" as a custom attribute.
+            let binding = (Def::NonMacroAttr(NonMacroAttrKind::Custom),
+                           ty::Visibility::Public, ident.span, Mark::root())
+                           .to_name_binding(self.arenas);
+            Ok(MacroBinding::Global(binding))
```

Comment:
To confirm, this basically means that it's impossible for an attribute to be "unresolved" now, right? All attributes that otherwise fail to get resolved simply get classified as "this is a custom attribute, it needs `#![feature(custom_attributes)]`"?

Reply:
>it's impossible for an attribute to be "unresolved" now

It's possible in some rare cases when import resolution/expansion is stuck due to indeterminacy, but otherwise yes, for a *single-segment* attribute you always get a custom attribute instead of "no resolution" (i.e. the old pre macro modularization behavior is restored).